Sources/Kitura/CodableRouter+TypeSafeMiddleware.swift

```diff
@@ -0,0 +1,1245 @@
+/*
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import LoggerAPI
+import KituraNet
+import KituraContracts
+
+// Type-safe middleware Codable router
+
+extension Router {
+
+    // MARK: Codable Routing With Type-Safe Middleware
```

Comment:
At least the With should be lowercase in the // MARK here as it looks strange with a capital 'W' underneath the previous MARK'd title of Parameter handling (which has a lowercase 'h').